src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +81,12 @@ public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+            .filter(person -> person.getSex() == Person.Sex.WOMAN && person.getAge() >= femaleAge)
+            .map(Person::getCats)
+            .flatMap(Collection::stream)
```

Comment:
here you can do both in one operation
```suggestion
         .flatMap(p -> p.getCats().stream())
```